Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm

```diff
@@ -346,13 +338,29 @@ static void updateCustomAppearance(CALayer *layer, GraphicsLayer::CustomAppearan
         node.setEventRegion(properties.eventRegion);
     updateMask(node, properties, relatedLayers);
 
-#if ENABLE(GAZE_GLOW_FOR_INTERACTION_REGIONS)
+#if ENABLE(GAZE_GLOW_FOR_INTERACTION_REGIONS) || HAVE(CORE_ANIMATION_SEPARATED_LAYERS)
     if (properties.changedProperties & LayerChange::VisibleRectChanged)
         node.setVisibleRect(properties.visibleRect);
+#endif
+
+#if ENABLE(GAZE_GLOW_FOR_INTERACTION_REGIONS)
     if (properties.changedProperties & LayerChange::EventRegionChanged || properties.changedProperties & LayerChange::VisibleRectChanged)
         updateLayersForInteractionRegions(node);
 #endif
 
+#if HAVE(CORE_ANIMATION_SEPARATED_LAYERS)
+    if (properties.changedProperties & LayerChange::SeparatedChanged)
+        node.setShouldBeSeparated(properties.isSeparated);
+
+    if (properties.changedProperties & LayerChange::SeparatedChanged || properties.changedProperties & LayerChange::VisibleRectChanged) {
```

Comment:
The visible rect code in WebCore is gated on `ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION)`, so I think you should generalize it to `ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION) | | HAVE(CORE_ANIMATION_SEPARATED_LAYERS)`